frontends/web/src/components/wait-dialog/wait-dialog.tsx

```diff
@@ -62,59 +44,56 @@ class WaitDialog extends Component<Props, State> {
     e.stopPropagation();
   };
 
-  private activate = () => {
-    this.setState({ active: true }, () => {
-      if (!this.overlay.current || !this.modal.current) {
-        return;
-      }
-      this.overlay.current.classList.add(style.activeOverlay);
-      this.modal.current.classList.add(style.activeModal);
-    });
+  const activate = () => {
+    if (!overlay.current || !modal.current) {
+      return;
+    }
+    overlay.current.classList.add(style.activeOverlay);
+    modal.current.classList.add(style.activeModal);
   };
 
-  public render() {
-    const {
-      t,
-      includeDefault,
-      title,
-      children,
-    } = this.props;
-    const defaultContent = (
-      <div>
-        <p className={style.confirmationLabel}>{t('confirm.info')}</p>
-      </div>
-    );
+  useEffect(() => {
+    document.body.addEventListener('keydown', handleKeyDown);
+    activate();
 
-    const hasChildren = React.Children.toArray(children).length > 0;
-    return (
-      <div
-        className={style.overlay}
-        ref={this.overlay}
-        style={{ zIndex: 10001 }}>
-        <UseDisableBackButton />
-        <div className={[style.modal, style.open].join(' ')} ref={this.modal}>
-          {
-            title && (
-              <div className={style.header}>
-                <h3 className={style.title}>{title}</h3>
-              </div>
-            )
-          }
-          <div className={style.contentContainer}>
-            <div className={style.content}>
-              { (hasChildren && includeDefault) ? defaultContent : null }
-              { hasChildren ? (
-                <div className="flex flex-column flex-start">
-                  {children}
-                </div>
-              ) : defaultContent }
+    return () => {
+      document.body.removeEventListener('keydown', handleKeyDown);
+    };
+  }, []);
+
+  const defaultContent = (
+    <div>
+      <p className={style.confirmationLabel}>{t('confirm.info')}</p>
+    </div>
+  );
+
+  const hasChildren = React.Children.toArray(children).length > 0;
+
+  return (
+    <div
+      className={style.overlay}
+      ref={overlay}
+      style={{ zIndex: 10001 }}>
+      <UseDisableBackButton />
+      <div className={[style.modal, style.open].join(' ')} ref={modal}>
+        {
+          title && (
+            <div className={style.header}>
+              <h3 className={style.title}>{title}</h3>
             </div>
+          )
+        }
+        <div className={style.contentContainer}>
+          <div className={style.content}>
+            { (hasChildren && includeDefault) ? defaultContent : null }
```

Comment:
We could probably simplify this In the future and remove `includeDefault` and defaultContent and move to the components that need it or have no children (BB01 components)